core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -972,6 +974,68 @@ protected Scope visitSetTableAuthorization(SetTableAuthorization node, Optional<
             return createAndAssignScope(node, scope);
         }
 
+        @Override
+        protected Scope visitTableExecute(TableExecute node, Optional<Scope> scope)
+        {
+            Table table = node.getTable();
+            QualifiedObjectName originalName = createQualifiedObjectName(session, table, table.getName());
+            if (metadata.getView(session, originalName).isPresent()) {
+                throw semanticException(NOT_SUPPORTED, node, "ALTER TABLE EXECUTE is not supported for views");
```

Comment:
also check materialized view (first; maintain same order as in `io.trino.sql.analyzer.StatementAnalyzer.Visitor#visitTable`, until  we do something like #9400)